docs/src/tutorial/particle_stateful.jl

```diff
@@ -0,0 +1,94 @@
+} <: AbstractParticleStateful{DIR,SPECIES,ELEMENT}
+    direction::DIR        # Incoming or outgoing
+    species::SPECIES      # Particle species (e.g., Muon, AntiMuon)
+    mom::ELEMENT          # Particle's four-momentum
+end
+
+# ## Implement the Interface Functions
+#
+# Next, we implement the required interface functions for `ExampleParticleStateful`, which are:
+#
+# - [`particle_direction`](@ref)
+# - [`particle_species`](@ref)
+# - [`momentum`](@ref)
+#
+# These functions extract the respective properties from the `ExampleParticleStateful` object.
+
+# Define the particle_direction function
+function particle_direction(part::ExampleParticleStateful)
+    return part.direction
+end
+
+# Define the particle_species function
+function particle_species(part::ExampleParticleStateful)
+    return part.species
+end
+
+# Define the momentum function
+function momentum(part::ExampleParticleStateful)
+    return part.mom
+end
+
+# ## Example Usage
+#
+# We can now create instances of `ExampleParticleStateful` for, say, `Electron` and `Positron`.
+using QEDcore  # to get predefined particles and four-momentum
+
+# Create a four-momentum vector (dummy example)
+momentum_electron = SFourMomentum(1.0, 0.0, 0.0, 0.0);  # E, px, py, pz
+
+# Create an incoming Electron using ExampleParticleStateful
+incoming_electron = ExampleParticleStateful(Incoming(), Electron(), momentum_electron);
+
+# Create an outgoing Positron using ExampleParticleStateful
+outgoing_positron = ExampleParticleStateful(Outgoing(), Positron(), momentum_electron);
+
+# Access particle properties
+println("Incoming electron mass: ", mass(particle_species(incoming_electron)))
+println(
+    "Is the outgoing positron a fermion? ", is_fermion(particle_species(outgoing_positron))
+)
+
+# Access momentum
+println("Incoming electron momentum: ", momentum(incoming_electron))
+println("Outgoing positron momentum: ", momentum(outgoing_positron))
+
+# ## Summary
+#
+# In this tutorial, we created a general `ExampleParticleStateful` type that can represent any particle species (like `Electron`,`Positron` from `QEDcore`, but also `Muon` or `AntiMuon` implemented in [this tutorial](@ref tutorial_particle)) by using the species as a type parameter. This approach avoids the need to define separate stateful types for each particle, making the implementation more flexible and reusable.
+#
+# The key steps were:
+#
+# 1. **Defining the general `ExampleParticleStateful` type** with parameters for direction, species, and momentum.
+# 2. **Implementing the interface functions** to retrieve direction, species, and momentum.
+# 3. **Using the generalized type** for all subtypes of `AbstractParticleType`.
+#
+# This method makes it easy to add more particles in the future while keeping the same structure.
+# !!! note "Reference implementation"
+#
+#     A reference implementation of a stateful particle type is given by
+#     [`QEDcore.ParticleStateful`](@extref)
```

Comment:
This links to the function (the constructor), not the type. They obviously have identical names, but I think it's possible to disambiguate.

Reply:
You are right, of course. A quick internet lookup gave no insights on how to disambiguate this. Would you happen to know a way?

Reply:
https://juliadocs.org/DocumenterInterLinks.jl/stable/write_inventory/#The-Julia-Domain
I think something like `(@extref jl:type:QEDcore.ParticleStateful)` should work but I haven't tried it myself yet.

Reply:
Thanks for the suggestion. Done in 97e4caa